base_decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```

Comment:
- [ ] As specified in the project requirement, the `BaseDecorator` is supposed to take the class `Neamable` as param and return the correct name like so:

```rb
def correct_name
      @nameable.correct_name
    end
```